website/cue/reference/components/sources/envoy_als.cue

```diff
@@ -0,0 +1,154 @@
+package metadata
+
+components: sources: envoy_als: {
+	_grpc_port: 9999
+
+	title: "Envoy ALS"
+
+	classes: {
+		commonly_used: false
+		delivery:      "at_least_once"
+		deployment_roles: ["daemon", "aggregator"]
+		development:   "beta"
+		egress_method: "stream"
+		stateful:      false
+	}
+
+	features: {
+		acknowledgements: false
+		multiline: enabled: false
+		receive: {
+			from: {
+				service: services.envoy
+
+				interface: socket: {
+					direction: "incoming"
+					port:      _grpc_port
+					protocols: ["tcp"]
+					ssl: "optional"
+				}
+			}
+			tls: {
+				// enabled per listener below
+				enabled: false
+			}
+		}
+	}
+
+	support: {
+		requirements: []
+		warnings: [
+			"""
+				The `envoy_als` source only supports http log events at this time.
```

Comment:
Yes, it's dependent on how the user configures Envoy. Essentially Envoy has support to be configured as L4 or L7 proxy. In the TCP case, no L7 protocol is assumed and you get TCP connection logs. If Envoy is used as an HTTP LB, you will get HTTP request logs sent rather than TCP logs.

TCP log support should be able to be added in a follow up without too much difficulty.

Reply:
Thanks for the extra context. Am I correct in understanding then that based on the user configuration, either TCP logs or HTTP logs will be emitted, and not in combination? If that is the case, I'm thinking now that we perhaps should just error out of the source at that point because it won't be able to ingest any events and will just be logging a bunch of warnings. What do you think?

Reply:
You could have many different Envoys connecting to 1 vector instance. In such a case, if several were configured in HTTP mode, they'd be working fine. If one was in TCP mode, you generate warnings, but I think we wouldn't want to stop all logs from coming in by taking down the source?

Reply:
Ah yes, excellent point.